src/Algebra/Construct/Add/Point.agda

```diff
@@ -0,0 +1,102 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definition of algebraic structures we get from adding a disjoint point
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Construct.Add.Point where
+
+open import Algebra.Bundles
+open import Algebra.Core using (Op₂)
+import Algebra.Definitions as Definitions
+import Algebra.Structures as Structures
+open import Data.Maybe.Relation.Binary.Pointwise
+open import Data.Product using (_,_)
+open import Level using (Level; _⊔_)
+open import Relation.Binary.Core
+open import Relation.Binary.Definitions
+open import Relation.Binary.Structures
+open import Relation.Nullary.Construct.Add.Point
+
+private
+  variable
+    a ℓ : Level
+
+lift-op : ∀ {A : Set a} → Op₂ A → Op₂ (Pointed A)
+lift-op op [ p ] [ q ] = [ op p q ]
+lift-op _  [ p ] ∙     = [ p ]
+lift-op _  ∙     [ q ] = [ q ]
+lift-op _  ∙     ∙     = ∙
+
+module _ {A : Set a} {_≈_ : Rel A ℓ} {op : Op₂ A} (refl-≈ : Reflexive _≈_) where
+  open Definitions
+  private
+    _≈∙_ = Pointwise _≈_
+
+  lift-≈ : ∀ {x y : A} → x ≈ y → Pointwise _≈_ [ x ] [ y ]
+  lift-≈ = just
+
+  lift-op-cong₂ : Congruent₂ _≈_ op → Congruent₂ _≈∙_ (lift-op op)
+  lift-op-cong₂ R-cong (just eq-l) (just eq-r) = lift-≈ (R-cong eq-l eq-r)
+  lift-op-cong₂ R-cong (just eq) nothing = lift-≈ eq
+  lift-op-cong₂ R-cong nothing (just eq) = lift-≈ eq
+  lift-op-cong₂ R-cong nothing nothing = refl refl-≈
+
+  lift-op-assoc : Associative _≈_ op → Associative _≈∙_ (lift-op op)
+  lift-op-assoc assoc [ p ] [ q ] [ r ] = lift-≈ (assoc p q r)
+  lift-op-assoc _     [ p ] [ q ] ∙     = refl refl-≈
+  lift-op-assoc _     [ p ] ∙     [ r ] = refl refl-≈
+  lift-op-assoc _     [ p ] ∙     ∙     = refl refl-≈
+  lift-op-assoc _     ∙     [ r ] [ q ] = refl refl-≈
+  lift-op-assoc _     ∙     [ q ] ∙     = refl refl-≈
+  lift-op-assoc _     ∙     ∙     [ r ] = refl refl-≈
+  lift-op-assoc _     ∙     ∙     ∙     = refl refl-≈
+
+  lift-op-identityˡ : LeftIdentity _≈∙_ ∙ (lift-op op)
+  lift-op-identityˡ [ p ] = refl refl-≈
+  lift-op-identityˡ ∙     = refl refl-≈
+
+  lift-op-identityʳ : RightIdentity _≈∙_ ∙ (lift-op op)
+  lift-op-identityʳ [ p ] = refl refl-≈
+  lift-op-identityʳ ∙     = refl refl-≈
+
+  lift-op-identity : Identity _≈∙_ ∙ (lift-op op)
+  lift-op-identity = lift-op-identityˡ , lift-op-identityʳ
+
+module _ {A : Set a} {_≈_ : Rel A ℓ} {op : Op₂ A} where
+  open Structures
+  private
+    _≈∙_ = Pointwise _≈_
+
+  lift-op-isMagma : IsMagma _≈_ op → IsMagma _≈∙_ (lift-op op)
+  lift-op-isMagma M =
+    record
+    { isEquivalence = isEquivalence M.isEquivalence
+    ; ∙-cong        = lift-op-cong₂ (IsEquivalence.refl M.isEquivalence) M.∙-cong
```

Comment:
I think all of these `IsEquivalence.refl M.isEquivalence` here and below can just be `M.refl`?